src/Microsoft.AspNetCore.Mvc.Core/Internal/PrefixContainer.cs

```diff
@@ -173,45 +173,7 @@ private static void GetKeyFromNonEmptyPrefix(string prefix, string entry, IDicti
             }
         }
 
-        public static bool IsPrefixMatch(string prefix, string testString)
```

Comment:
Minor but this version looks slightly cleaner (except for the `switch` statement). Suggest copying comments like `// not long enough` and `// prefix doesn't match`. Also good to make the `StartsWith()` part `return` early.